finetune.py

```diff
@@ -0,0 +1,302 @@
+    optimizer, lr_scheduler = utils.get_optim(args.lr, total_steps, model)
+
+    wandb_run = None
+    # Logger instantiation/configuration
+    if args.wandb:
+        import wandb
+
+        logging.info("Instantiating WandbLogger.")
+        wandb_run = utils.init_wandb_from_config(job_type="finetuning")
+
+        wandb.define_metric("global_step")
+        wandb.define_metric("epochs")
+        wandb.define_metric("train_step/*", step_metric="global_step")
+        wandb.define_metric("learning_rates/*", step_metric="global_step")
+        wandb.define_metric("finetune/*", step_metric="epochs")
+
+    loader_args = dict(
+        dataset=args.dataset,
+        path=args.data_path,
+        num_workers=args.num_workers,
+        batch_size=args.batch_size,
+        target_config={"graph": ["energy", "stress"], "node": ["forces"]},
+    )
+    train_loader = data_loaders.build_train_loader(
+        **loader_args,
+        augmentation=True,
+    )
+    logging.info("Starting training!")
+
+    num_steps = args.num_steps
+
+    start_epoch = 0
+
+    for epoch in range(start_epoch, args.max_epochs):
+        print(f"Start epoch: {epoch} training...")
+        avg_train_metrics = finetune(
+            model=model,
+            optimizer=optimizer,
+            dataloader=train_loader,
+            lr_scheduler=lr_scheduler,
+            clip_grad=args.gradient_clip_val,
+            device=device,
+            num_steps=num_steps,
+            epoch=epoch,
+        )
+
+        if args.wandb:
+            wandb.run.log(
+                utils.prefix_keys(avg_train_metrics, "finetune"), commit=False
+            )
+            wandb.run.log({"epoch": epoch}, commit=True)
+
+        # Save checkpoint from last epoch
+        if epoch == args.max_epochs - 1:
+            checkpoint = {
+                "epoch": epoch,
+                "model_state_dict": model.state_dict(),
+                "optimizer_state_dict": optimizer.state_dict(),
+                "lr_scheduler_state_dict": (
+                    lr_scheduler.state_dict() if lr_scheduler else None
+                ),
+            }
+            torch.save(
+                checkpoint,
+                os.path.join(args.checkpoint_path, f"checkpoint_epoch{epoch}.ckpt"),
+            )
+            logging.info(f"Checkpoint saved to {args.checkpoint_path}")
+
+    if wandb_run is not None:
+        wandb_run.finish()
+
+
+def main():
+    """Main."""
+    parser = argparse.ArgumentParser(
+        description="Finetune orb model",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+    parser.add_argument(
+        "--random_seed", default=1234, type=int, help="Random seed for finetuning."
+    )
+    parser.add_argument(
+        "--wandb",
+        default=True,
+        action="store_true",
+        help="If the run is logged to wandb.",
+    )
+    parser.add_argument("--dataset", default="mp-traj", type=str, help="Dataset name.")
```

Comment:
Explain what impact this has / how it is used

Reply:
Yeah the dataset argument was for the `AseSqliteDataset`clas but I realize it is not necessary to have this name, path is enough. Now the dataset is passed to wandb as part of the run name for logging purpose.